pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +92,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthPlugin is a custom plugin used to authenticate to the kubernetes cluster.
+	AuthPlugin string `json:"auth-plugin,omitempty"`
```

Comment:
> Other than figuring out naming, is this something we can punt on until we have a concrete use case? Or does the fact that this gets serialized mean we need to figure it out up front?

I think we should decide which fields get plumbed through to the plugins before deciding on a serialization.  Passing the information from the `AuthInfo` doesn't seem unreasonable.  It also doesn't seem unreasonable to allow plumb through a `map[string]string` or an object registered in the scheme down through either.